scripts.js

```diff
@@ -266,19 +258,23 @@ function handleDropdownRegion() {
     fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
   }
   
-  // Get currently selected region if in the same region page based on localeHome value
-  const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
-  if (!currentRegion.length) {
-    // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use language
-    sessionStorage.getItem('blog-selected-language') || window.blog.language;
-  } else {
-      // uk and apac, just save the info in the session storage
-      sessionStorage.setItem('blog-selected-language', currentRegion[0].lang);
-  }
+  // get actual selected Region
+  getActualRegion = function() {
+    const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
```

Comment:
I would use `find()` instead of `filter()` here, since you're only interested in the first result anyway. This will simplify the code further down as well.